### DISCUSSION TITLE
Is it possible to record a video only for the last retry if a previous failure happend?

### DISCUSSION THREAD

**Discussion by rafawalter** [cypress-io/cypress#26629] [Questions and Help]
Recording videos is very useful for resolving failed tests.

Still, they are heavy on disk and processor.

Is it possible to configure Cypress so it records videos only for the last retry if a previous failure happend in the same test?

Maybe changing the video config based on the current attempt counter...
https://docs.cypress.io/guides/guides/test-retries#Can-I-access-the-current-attempt-counter-from-the-test

**Comment by portyu9** [comment]
Hi @rafawalter 

There isn't a built-in way to enable video recording only when a test reaches its last retry.

Cypress video recording is done at the **spec level**, not independently for each test attempt. When `video: true` is enabled, Cypress records the spec while it runs, including its retry attempts.

Cypress does support a useful related optimization: in the `after:spec` event you can delete the video when the spec had **no failures and no retries**. Cypress will then skip processing/compressing/uploading that video to Cypress Cloud.

That saves storage and post-processing/upload work, but it does **not** save the CPU/disk cost of capturing the video during the test run in the first place.

So if the main problem is the recording overhead itself, the choices are essentially to disable video recording or keep it enabled for the whole spec; it cannot currently be switched on only for the final retry.

Please mark this answer as accepted if it helped, thank you.

**Reply by rafawalter** [reply]
Hi, thanks for the reply.

You got it right, the main issue is the recording overhead, processing and disk space and CI time.

Yep, we disable video recording. But having them would be helpful to fix issues detected with Cypress. Just like the screenshots do. Recording videos only when we repeat failed tests would be awesome!

Unfortunately it's hard for us to get Cypress Cloud. The open source version is really helpful.

Thanks for your help

Rafael

**Reply by portyu9** [reply]
Agreed @rafawalter
Please mark this answer as accepted if it helped, thank you.
